sig-api/charter.md

```diff
@@ -0,0 +1,22 @@
+# SIG API
+
+Weekly meetings on Wednesdays at 13:00 UTC.
+https://docs.google.com/document/d/1LCLt5k9x1yISZE_0SxGb5hcU3Q0pzDqRfIMEffgqlnk/edit?usp=sharing
+
+## Scope
```

Comment:
While sig-compute will focus on the node level aspects, Sig-api will need to be responsible for the kubevirt control plane components and deployment such as virt-api, virt-conrtoller, and virt-operator.

Reply:
As I see it, the scope is not about a specific logic or features content, but on how to expose them for usage through the API in a safe and stable manner.

Could you be specific and point out which of the items listed here requires rephrasing, removing or which items are missing?

At the moment, I see that all of the items are described in a supportive manner. I mean, trying to provide means to write safe and stable APIs.

Reply:
Yes, let me second @EdDev .

IIUIC - and how I would support it - SIG API is exclusively about guarding our public APIs (everything in CRDs)., and not about their implementation.

It is to ensure that we have
- good API patterns
- avoid repeating API mistakes
- achieve more consistency between APIs
- ensure proper life-cycling of APIs (incl FGs).

Thus I am in favor of scoping SIG API in that way.

@vladikr so far I'm seeing the components that you had mentioned still to be in the domain of sig compute.

Reply:
@fabiand I don't think that sig-compute knows about this responsibility. 
If you look at [Sig API Machinery,](https://github.com/kubernetes/community/blob/master/sig-api-machinery/charter.md) the focus there is on everything that affects the control plane.   

I see very little value and a risk for a bottleneck in a sig that is only responsible for writing APIs without the knowledge and responsibility of how these APIs affect the rest of the components.

Reply:
@vladikr , this proposal is not related to [api-machinery](https://github.com/kubernetes/community/blob/master/sig-api-machinery/charter.md), it is related to the [API governance subproject](https://github.com/kubernetes/community/tree/master/sig-architecture#architecture-and-api-governance-1) of [sig-architecture](https://github.com/kubernetes/community/blob/master/sig-architecture/charter.md).
It is actually a portion of that subproject, as it only focused on the API part.

> I see very little value and a risk for a bottleneck in a sig that is only responsible for writing APIs without the knowledge and responsibility of how these APIs affect the rest of the components.

The SIG is not responsable to write APIs, it is responsible to provide guidance and protect the project API, keeping it safe in terms of comparability, dependability and consistency.

IMO the project reached a size that makes it hard for all maintainers/approvers to consider all API concerns and avoid conflict of interest with other concerns (functionality, code quality, delivery due time).

I would expect the SIG-API to evaluate an API change and seek/ask for several alternatives before it is being accepted.
This may be considered a bottleneck, although this charter is not hinting anything about enforcing at this stage. I think it makes sense to be very careful when introducing API changes at this stage, it should be much harder than before, with more discussions taking place.
I think we are not yet in a good position today, although awareness has increased.

Reply:
To me +1 about the scope

This sig is *not implementing* APIs, it's rather focused on establishing standards around it - as described below.

Implementing, and maintaining APIs is still up to the other SIG (whoever is providing it), and the machinery is currently owned by sig compute (IIUIC) - but this is separate from this proposed sig.

Btw - one takeaway could be @EdDev @alaypatel07 to call this sig `sig-api-governance`.